src/sdk/namespace_fs.js

```diff
@@ -609,6 +609,7 @@ class NamespaceFS {
     }
 
     async _list_objects(params, object_sdk, list_versions) {
+        let list_unsorted = false;
```

Comment:
it seems that you forced this to list_objects v1, which we cannot really do because the clients assume that NextMarker is only returned in some cases of abiguity (when both objects and prefixes are returned), but otherwise in flat listing the clients typically have to assume that the marker is the Key of the last entry.

So I think we want to apply this only for list_type === '2'. Some of the code can be shared, but preferably not use the same _list_objects() method for both cases (instead _list_objects should have some common code and the call _list_objects_v1 or _list_objects_v2) because I think this is too messy to read, even now it is a long and hard function to maintain.